### PR TITLE
Added OAuth2 authentication

### PR DESCRIPTION
**Pull request by gfollesa**
Added OAuth2 authentication without OpenID

### CODE REVIEW

File: backend/src/Squidex/Config/Authentication/LoggingHandler.cs [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+using System;
+using System.IO;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Squidex.Config.Authentication
+{
+    public class LoggingHandler : DelegatingHandler
```
Comment (SebastianStehle) [inline backend/src/Squidex/Config/Authentication/LoggingHandler.cs:16]: What is this class? I do not understand the purpose.

File: backend/src/Squidex/Config/Authentication/OAuth2Handler.cs [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+            return Task.CompletedTask;
+        }
+
+        public override Task TicketReceived(TicketReceivedContext context)
+        {
+            var test = context.Request.Body;
+            return base.TicketReceived(context);
+        }
+
+        public override Task RemoteFailure(RemoteFailureContext context)
```
Comment (SebastianStehle) [inline backend/src/Squidex/Config/Authentication/OAuth2Handler.cs:33]: Why do you override it, when you do not have a custom implementation?

File: backend/src/Squidex/Config/Authentication/OAuth2Handler.cs [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+        }
+
+        public override Task AccessDenied(AccessDeniedContext context)
+        {
+            return base.AccessDenied(context);
+        }
+
+        public override async Task CreatingTicket(OAuthCreatingTicketContext context)
+        {
+            if (context.Options.UserInformationEndpoint != null)
```
Comment (SebastianStehle) [inline backend/src/Squidex/Config/Authentication/OAuth2Handler.cs:45]: What is this? Is this part of the oauth specification? If yes, why is not handled by the default handler?

File: backend/src/Squidex/Config/Authentication/OAuth2AuthenticationServices.cs [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    {
+        public static AuthenticationBuilder AddSquidexExternalOAuth2Authentication(this AuthenticationBuilder authBuilder, MyIdentityOptions identityOptions)
+        {
+            if (identityOptions.IsOAuthConfigured())
+            {
+                HttpClientHandler hch = new HttpClientHandler();
+                hch.ServerCertificateCustomValidationCallback = delegate { return true; };
+
+                HttpClient httpClient = new HttpClient(new LoggingHandler(hch));
+                httpClient.SetBasicAuthenticationOAuth(identityOptions.OAuthClientId, identityOptions.OAuthSecret);
```
Comment (SebastianStehle) [inline backend/src/Squidex/Config/Authentication/OAuth2AuthenticationServices.cs:26]: What is this? Is this part of the oauth specification? Why do you have to define it?

File: backend/src/Squidex/Config/Authentication/OAuth2AuthenticationServices.cs [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+
+namespace Squidex.Config.Authentication
+{
+    public static class OAuth2AuthenticationServices
+    {
+        public static AuthenticationBuilder AddSquidexExternalOAuth2Authentication(this AuthenticationBuilder authBuilder, MyIdentityOptions identityOptions)
+        {
+            if (identityOptions.IsOAuthConfigured())
+            {
+                HttpClientHandler hch = new HttpClientHandler();
```
Comment (SebastianStehle) [inline backend/src/Squidex/Config/Authentication/OAuth2AuthenticationServices.cs:22]: Please use "var" whenever possible

**Review by SebastianStehle — CHANGES_REQUESTED** [review]

File: backend/src/Squidex/Config/Authentication/LoggingHandler.cs (hunk 1 above)
Comment (gfollesa) [reply backend/src/Squidex/Config/Authentication/LoggingHandler.cs:16]: Sorry. This is a debug class. I'll remove it.

File: backend/src/Squidex/Config/Authentication/OAuth2Handler.cs (hunk 2 above)
Comment (gfollesa) [reply backend/src/Squidex/Config/Authentication/OAuth2Handler.cs:33]: Ok. I'll remove useless overrides

File: backend/src/Squidex/Config/Authentication/OAuth2Handler.cs (hunk 3 above)
Comment (gfollesa) [reply backend/src/Squidex/Config/Authentication/OAuth2Handler.cs:45]: It's a part of oauth specification if there's a REST service to retrieve user informations. Unfortunately it's not handled by the default handler.

File: backend/src/Squidex/Config/Authentication/OAuth2AuthenticationServices.cs (hunk 5 above)
Comment (gfollesa) [reply backend/src/Squidex/Config/Authentication/OAuth2AuthenticationServices.cs:22]: Ok.

File: backend/src/Squidex/Config/Authentication/OAuth2AuthenticationServices.cs (hunk 4 above)
Comment (gfollesa) [reply backend/src/Squidex/Config/Authentication/OAuth2AuthenticationServices.cs:26]: The basic authentication is a part of the oauth specification. It's optional, I'll parametrize it.

File: backend/src/Squidex/Config/Authentication/OAuth2Handler.cs (hunk 3 above)
Comment (SebastianStehle) [reply backend/src/Squidex/Config/Authentication/OAuth2Handler.cs:45]: In OpenID connect the user id claim is called "sub" ... is this no available in the token by default?

Comment (SebastianStehle) [reply backend/src/Squidex/Config/Authentication/OAuth2Handler.cs:45]: I like this approach with the claim actions: https://developer.okta.com/blog/2019/07/12/secure-your-aspnet-core-app-with-oauth

**Pull request closed without merging**